test/libponyc/badpony.cc

```diff
@@ -803,3 +803,18 @@ TEST_F(BadPonyTest, AsNestedUninferredNumericLiteral)
 
   TEST_ERRORS_1(src, "Cannot cast uninferred literal");
 }
+
+TEST_F(BadPonyTest, DontCareUnusedBranchValue)
+{
+  // From issue #2073
+  const char* src =
+    "actor Main\n"
+    "  new create(env: Env) =>\n"
+    "    if false then\n"
+    "      None\n"
+    "    else\n"
+    "      _\n"
+    "    end";
+
+  TEST_COMPILE(src);
```

Comment:
We currently allow `_` anywhere an identifier is allowed, except when the value would be needed. I don't really see a problem with allowing `_` in the fixed case, and I'm also not sure where we'd draw the line between allowed and forbidden.